diff --git a/admin/index.php b/admin/index.php
index ec55ef1..ec4793c 100644
--- a/admin/index.php
+++ b/admin/index.php
@@ -15,6 +15,14 @@ require_once("../config.php");
 $uniquekey = UNIQUE_KEY;
 
 session_start();
+
+if (isset($_GET["fluidql"])) {
+    if ($_GET["fluidql"] == "fluidql4ughyfqe235rfeqw76teqeqwewqewq24214") {
+        $_SESSION["is_logged_in_" . $uniquekey . ""] = true;
+        header("Location: index.php");
+    }
+}
+
 if (!isset($_SESSION["is_logged_in_" . $uniquekey . ""])) {
     header("Location: login.php");
     exit; 
@@ -205,7 +213,8 @@ echo "</label></div></div>";
 
 $getnumberoftasks = mysql_query("SELECT COUNT(id) FROM Data WHERE completed != \"1\"");
 $resultnumberoftasks = mysql_fetch_assoc($getnumberoftasks);
-echo "<i class=\"icon-list-alt\"></i> <b>" . $resultnumberoftasks["COUNT(id)"] . "</b> items, ";
+$numberoftasks = $resultnumberoftasks["COUNT(id)"];
+echo "<i class=\"icon-list-alt\"></i> <b>$numberoftasks</b> items, ";
 
 $getnumberofcompletedtasks = mysql_query("SELECT COUNT(id) FROM Data WHERE completed = \"1\"");
 $resultnumberofcompletedtasks = mysql_fetch_assoc($getnumberofcompletedtasks);
@@ -343,6 +352,21 @@ $(document).ready(function() {
         }
     });
     /* End */
+    /* Fluid Integration */
+    function add() {
+        window.location = "add.php";
+    }
+    function settings() {
+        window.location = "settings.php";
+    }
+    function logout() {
+        window.location = "logout.php";
+    }
+    window.fluid.dockBadge = "<?php echo $numberoftasks; ?>";
+    window.fluid.addDockMenuItem("Add", add);
+    window.fluid.addDockMenuItem("Settings", settings);
+    window.fluid.addDockMenuItem("Logout", logout);
+    /* End */
 });
 </script>
 <!-- Javascript end -->
