diff --git a/admin/index.php b/admin/index.php
index 8f573bb..4747664 100644
--- a/admin/index.php
+++ b/admin/index.php
@@ -4,7 +4,7 @@
 
 $version = "1.0";
 $codename = "BarredBuck";
-$rev = "20";
+$rev = "30";
 
 if (!file_exists("../config.php")) {
     header("Location: ../installer");
@@ -96,10 +96,25 @@ if (!$does_db_exist) {
     die("<div class=\"alert alert-error\"><h4 class=\"alert-heading\">Error</h4><p>Database does not exist (" . mysql_error() . "). Check your database settings are correct.</p><p><a class=\"btn btn-danger\" href=\"javascript:history.go(-1)\">Go Back</a></p></div></div></body></html>");
 }
 
-if (!isset($_GET["showcompleted"])) {
-    $gettasks = mysql_query("SELECT * FROM Data WHERE completed != \"1\"");
+$cat = $_GET["cat"];
+
+if (isset($_GET["cat"])) {
+    $checkcat = mysql_query("SELECT category FROM Data WHERE category = \"$cat\"");
+    $resultcheckcat = mysql_fetch_assoc($checkcat); 
+    if ($resultcheckcat == 0) {
+        die("<div class=\"alert alert-error\"><h4 class=\"alert-heading\">Error</h4><p>Category does not exist.</p><p><a class=\"btn btn-danger\" href=\"javascript:history.go(-1)\">Go Back</a></p></div></div></body></html>");
+    }
+    if (!isset($_GET["showcompleted"])) {
+        $gettasks = mysql_query("SELECT * FROM Data WHERE completed != \"1\" AND category = \"$cat\"");
+    } else { 
+        $gettasks = mysql_query("SELECT * FROM Data WHERE category = \"$cat\"");
+    }
 } else {
-    $gettasks = mysql_query("SELECT * FROM Data");
+    if (!isset($_GET["showcompleted"])) {
+        $gettasks = mysql_query("SELECT * FROM Data WHERE completed != \"1\"");
+    } else {
+        $gettasks = mysql_query("SELECT * FROM Data");
+    }  
 }
 
 //Update checking
@@ -172,6 +187,7 @@ while($row = mysql_fetch_assoc($gettasks)) {
 echo "</tbody></table>";
 
 ?>
+<div class="btn-toolbar">
 <div class="btn-group">
 <button id="edit" class="btn">Edit</button>
 <button id="delete" class="btn">Delete</button>
@@ -184,8 +200,22 @@ if (isset($_GET["showcompleted"])) {
 
 ?>
 </div>
-<br>
-<br>
+<div class="btn-group">
+<a class="btn dropdown-toggle" data-toggle="dropdown" href="#">Filter Category:<span class="caret"></span></a>
+<ul class="dropdown-menu">
+<?php
+
+$getcategories = mysql_query("SELECT DISTINCT(category) FROM Data WHERE category != \"\"");
+
+echo "<li><a href=\"index.php\">No filter</a></li>";
+while($row = mysql_fetch_assoc($getcategories)) {    
+    echo "<li><a href=\"?cat=" . $row["category"] . "\">" . ucfirst($row["category"]) . "</a></li>";
+}
+
+?>    
+</ul>
+</div>
+</div>
 <div class="alert alert-info">   
 <strong>Info:</strong> Important tasks are highlighted yellow, completed tasks green and overdue tasks red.  
 </div>
